packages/select/src/components/query-list/queryList.tsx

```diff
@@ -231,6 +269,28 @@ export class QueryList<T> extends React.Component<IQueryListProps<T>, IQueryList
         }
     }
 
+    private defaultMenuRenderer = (listProps: IItemListRendererProps<T>) => {
+        const { initialContent, noResults } = this.props;
+        const menuContent = QueryList.renderFilteredItems(listProps, noResults, initialContent);
+        return <Menu ulRef={listProps.itemsParentRef}>{menuContent}</Menu>;
+    };
+
+    private renderItem = (item: T, index?: number) => {
+        const { activeItem, query } = this.props;
+        const matchesPredicate = this.state.filteredItems.indexOf(item) >= 0;
```

Comment:
I know this isn't new, but wouldn't it have been better in hindsight to cache the filtered items as an ES6 Set (or anything that's not O(N), really) to speed up this lookup?

Reply:
yes, definitely a pain point in my head. we should do such a refactor.